# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../gcc/stdc++-cstddef-cross.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

During cross builds g++ does not find the toolchain stddef.h header, as the
include dir was scanned before including the cstddef header. Also my
understanding of include_next is to include the next, simillar named file -
not some completely different file.

  - Rene Rebe <rene@exactcode.de>

--- gcc-4.3.1/libstdc++-v3/include/c/cstddef.vanilla	2008-08-13 13:48:12.000000000 +0200
+++ gcc-4.3.1/libstdc++-v3/include/c/cstddef	2008-08-13 13:49:11.000000000 +0200
@@ -40,6 +40,6 @@
 #define __need_ptrdiff_t
 #define __need_NULL
 #define __need_offsetof
-#include_next <stddef.h>
+#include <stddef.h>
 
 #endif
--- gcc-4.3.1/libstdc++-v3/include/c_global/cstddef.vanilla	2008-08-13 13:48:38.000000000 +0200
+++ gcc-4.3.1/libstdc++-v3/include/c_global/cstddef	2008-08-13 13:48:56.000000000 +0200
@@ -45,7 +45,7 @@
 #pragma GCC system_header
 
 #include <bits/c++config.h>
-#include_next <stddef.h>
+#include <stddef.h>
 
 #ifndef _GLIBCXX_CSTDDEF
 #define _GLIBCXX_CSTDDEF 1
--- gcc-4.3.1/libstdc++-v3/include/c/cstdarg.vanilla	2008-08-18 15:10:20.000000000 +0200
+++ gcc-4.3.1/libstdc++-v3/include/c/cstdarg	2008-08-18 15:10:23.000000000 +0200
@@ -37,6 +37,6 @@
 #pragma GCC system_header
 
 #undef __need___va_list
-#include_next <stdarg.h>
+#include <stdarg.h>
 
 #endif
--- gcc-4.3.1/libstdc++-v3/include/c_global/cstdarg.vanilla	2008-08-18 15:09:38.000000000 +0200
+++ gcc-4.3.1/libstdc++-v3/include/c_global/cstdarg	2008-08-18 15:09:42.000000000 +0200
@@ -45,7 +45,7 @@
 #pragma GCC system_header
 
 #include <bits/c++config.h>
-#include_next <stdarg.h>
+#include <stdarg.h>
 
 #ifndef _GLIBCXX_CSTDARG
 #define _GLIBCXX_CSTDARG 1
